include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -865,6 +865,25 @@ struct mix_helper<T, T NBL_PARTIAL_REQ_BOT(VECTOR_SPECIALIZATION_CONCEPT) >
 	}
 };
 
+template<typename T, typename U>
+NBL_PARTIAL_REQ_TOP(VECTOR_SPECIALIZATION_CONCEPT&& concepts::Boolean<U>&& !(vector_traits<T>::Dimension == vector_traits<U>::Dimension) && concepts::BooleanScalar<U>)
+struct mix_helper<T, U NBL_PARTIAL_REQ_BOT(VECTOR_SPECIALIZATION_CONCEPT&& concepts::Boolean<U>&& !(vector_traits<T>::Dimension == vector_traits<U>::Dimension) && concepts::BooleanScalar<U>) >
```

Comment:
we have 6 partial specs of `mix_helper` we should clean them up

we can use named concepts/checkers suchs as
- `spirv::SelectIsCallable<T,U>` for HLSL
- `spirv::FMixIsCallable<T>` for HLSL
- `impl::glm::MixIsCallable` for C++

And a final one thats `MixCallingBuiltins<T,U>` which tells you if any of the above could have been used.

Then redo the last 3 unified C++/HLSL specializations (`T,T`, `T,U` with U scalar boolean and boolean vector) in terms of beginning with  `!MixCallingBuiltins<T,U> && (clear simple conditions here)`

So the code is more legible

Continuation of https://github.com/Devsh-Graphics-Programming/Nabla/pull/811#discussion_r2182835143